tests/data/test_minute_bars.py

```diff
@@ -451,6 +452,44 @@ def test_pad_data(self):
 
         self.assertEquals(100.0, volume_price)
 
+    def test_nans(self):
+        """
+        Test writing empty data.
+        """
+        sid = 1
+        last_date = self.writer.last_date_in_output_for_sid(sid)
+        self.assertIs(last_date, NaT)
+
+        self.writer.pad(sid, TEST_CALENDAR_START)
+
+        last_date = self.writer.last_date_in_output_for_sid(sid)
+        self.assertEqual(last_date, TEST_CALENDAR_START)
+
+        freq = self.market_opens.index.freq
+        minute = self.market_opens[TEST_CALENDAR_START + freq]
+        minutes = date_range(minute, minute + Timedelta('8 min'), freq='min')
+        data = DataFrame(
+            data={
+                'open': [nan, nan, nan, nan, nan, nan, nan, nan, nan],
```

Comment:
One of the likely causes of our upstream bug was that there are many different bit-representations of `NaN`.  (c.f http://steve.hollasch.net/cgindex/coding/ieeefloat.html)  It might be a good idea to verify that we correctly convert any bit representation of `NaN`.

We could get different NaNs via something like:

```
In [11]: ((0b11111111111 << 52) + arange(1, 11, dtype=int64)).view(float64)
Out[11]: array([ nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan,  nan])
```